pkg/workload/tpcc/payment.go

```diff
@@ -140,13 +143,68 @@ func createPayment(ctx context.Context, config *tpcc, mcp *workload.MultiConnPoo
 		VALUES ($1, $2, $3, $4, $5, $6, $7, $8)`,
 	)
 
+	if p.config.isLongDurationWorkload {
+		p.resetWorkhouse = p.sr.Define(`
+		UPDATE warehouse
+		SET w_ytd = $1
+		WHERE w_id >= $2 AND w_id < $3`,
+		)
+
+		p.resetDistrict = p.sr.Define(`
+		UPDATE district
+		SET d_ytd = $1
+		WHERE d_w_id >= $2 AND d_w_id < $3`,
+		)
+
+		// Starting the background goroutine which will reset the w_ytd values periodically in warehouseWytdResetPeriod
+		go p.startResetValueWorker(ctx)
+	}
+
 	if err := p.sr.Init(ctx, "payment", mcp); err != nil {
 		return nil, err
 	}
 
 	return p, nil
 }
 
+func (p *payment) startResetValueWorker(ctx context.Context) {
+	ticker := time.NewTicker(warehouseWytdResetPeriod)
+	defer ticker.Stop()
+
+	for {
+		select {
+		case <-ctx.Done():
```

Comment:
Thanks @itsbilal for the pointer.
I have implemented the wait group since I think it might be a cleaner solution if in future we might want to add for background reset tasks.